scipy/interpolate/_cubic.py

```diff
@@ -642,149 +643,153 @@ def __init__(self, x, y, axis=0, bc_type='not-a-knot', extrapolate=None):
             else:
                 extrapolate = True
 
-        dxr = dx.reshape([dx.shape[0]] + [1] * (y.ndim - 1))
-        slope = np.diff(y, axis=0) / dxr
-
-        # If bc is 'not-a-knot' this change is just a convention.
-        # If bc is 'periodic' then we already checked that y[0] == y[-1],
-        # and the spline is just a constant, we handle this case in the same
-        # way by setting the first derivatives to slope, which is 0.
-        if n == 2:
-            if bc[0] in ['not-a-knot', 'periodic']:
-                bc[0] = (1, slope[0])
-            if bc[1] in ['not-a-knot', 'periodic']:
-                bc[1] = (1, slope[0])
-
-        # This is a very special case, when both conditions are 'not-a-knot'
-        # and n == 3. In this case 'not-a-knot' can't be handled regularly
-        # as the both conditions are identical. We handle this case by
-        # constructing a parabola passing through given points.
-        if n == 3 and bc[0] == 'not-a-knot' and bc[1] == 'not-a-knot':
-            A = np.zeros((3, 3))  # This is a standard matrix.
-            b = np.empty((3,) + y.shape[1:], dtype=y.dtype)
-
-            A[0, 0] = 1
-            A[0, 1] = 1
-            A[1, 0] = dx[1]
-            A[1, 1] = 2 * (dx[0] + dx[1])
-            A[1, 2] = dx[0]
-            A[2, 1] = 1
-            A[2, 2] = 1
-
-            b[0] = 2 * slope[0]
-            b[1] = 3 * (dxr[0] * slope[1] + dxr[1] * slope[0])
-            b[2] = 2 * slope[1]
-
-            s = solve(A, b, overwrite_a=True, overwrite_b=True,
-                      check_finite=False)
-        elif n == 3 and bc[0] == 'periodic':
-            # In case when number of points is 3 we should count derivatives
-            # manually
-            s = np.empty((n,) + y.shape[1:], dtype=y.dtype)
-            t = (slope / dxr).sum() / (1. / dxr).sum()
-            s.fill(t)
+        if y.size == 0:
+            # bail out early for zero-sized arrays
+            s = np.zeros_like(y)
```

Comment:
Can you confirm that the diff just looks deceptively large here, but none of the subsequent code has changed apart from indentation due to the extra conditional? Maybe that would make the code review a bit less intimidating?

Reply:
Yup - you can see by hiding whitespace changes:
![image](https://user-images.githubusercontent.com/6570539/202877649-e96f5c19-875c-4ed4-822c-8433a17d5e37.png)

https://github.com/scipy/scipy/pull/17305/files?diff=unified&w=1

Reply:
Wow, this is great to know, Matt! 

And indeed, the diff here is nothing but an extra level of indentation + rewrap long lines because  the 79 characters mark.